src/System.Slices/src/System/Contract.cs

```diff
@@ -31,17 +31,26 @@ public static void Requires(bool condition)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void RequiresInRange(int start, int length)
         {
-            if (!(start >= 0 && start < length))
+            // explicit unchecked block is used in order to avoid OverflowException when casting 
+            // negative int to unsigned int when the code is compiled with /checked option
+            unchecked
             {
-                throw new ArgumentOutOfRangeException();
+                // use unsigned int to reduce range check by one ( >= 0 )
+                if ((uint)start >= (uint)length)
```

Comment:
Asserting the value is the least we should be doing here, but asserts are generally compiled out and this is a `public` entry-point. Seems dangerous to me. I'm not sure you can get away from a two part test scenario. `length >= 0 && (uint)start < (uint)length`.

There are many days when I wish CLS didn't care about signed-ness.